set.mm

```diff
@@ -382517,10 +382525,20 @@ proved using the inner form (see theorem 9.6 of [Schwabhauser] p. 69
 
   ${
     $d a b g i j $.
-    $( Define the relation congruence bewteen shapes.  Definition 4.4 of
-       [Schwabhauser] p. 35.  Ideally, we would define this for functions of
-       any set, but we will used words (functions over ` NN ` ) in most cases.
-       (Contributed by Thierry Arnoux, 3-Apr-2019.) $)
+    $( Define the relation congruence between shapes.  Definition 4.4 of
+       [Schwabhauser] p. 35.  Two shapes are congruent if all corresponding
+       segments between all corresponding points are congruent.  Ideally, we
+       would define this for functions of any set, but we will use words
```

Comment:
It looks like the current definition sees shapes as partial functions on RR.

Reply:
The reason for using functions on `RR` was that it could be used to define congruence of more complex shapes, like maybe segments and circles (provided they have the same parametrization. In fact until now, the definition has only been used on triangles represented by their 3 vertices.

Reply:
So why not take any common domain ? Or in the other direction (probably to be preferred), why not really take finite sequences i.e. `( 0 ..^ n )` for some `n e. _om` ? That way, you have a theory independent from RR, which is in the spirit of Tarski's FOL axiomatization of geometry (I think models are powers of real closed fields, not necessarily RR ?).

Other question: can you point me to the theorem that says that if the triples (abc) and (def) are congruent and if Babc then Bdef ? That theorem is the crucial one, it even justifies the phrase "congruent" applied to shapes. And from that, you of course obtain angle congruence.

(The idea is that "congruence" of two things means that all primitive notions agree on these two things. And obviously, then all derived notions also agree.)

Reply:
Can you suggest a better wording?

Reply:
@david-a-wheeler : if the question is to me: no, I cannot ;)  I think the decision is in @tirix's hands and then he/we'll adapt the text to the definition he chooses.
In any case, I do not know why "ideally" is used: if it were ideal, then it is very easy to make the change. On the other hand, I have not practiced Tarski's geometry enough to know what is gained in considering infinite (and even non-discrete) shapes. Considering infinite shapes removes the feel of simplicity of Tarski's geometry.

Reply:
The "Ideally..." text was in the original, I didn't write that (though it makes sense to me). If we don't have any specific changes to that sentence we want to make, I propose we just keep it as-is until we do :-).

Reply:
Let me paste Schwabäuser's definition below:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/5831830/216841767-bf6f2666-bfb4-4a1e-bb22-958ed162c308.png">
Which I would translate to:
> To describe more conveniently several congruences, we introduce the congruence of "geometrical figures" (made of finitely many points)
> 
> Definition: The n-tuples of points $< a_1,...,a_n >$ and $< b_1,...,b_n >$ are said to be congruent when: ... 

Word like `<" A B C ">` are the convenient way in Metamath to express tuples, so if we prefer to restrict ourselves strictly to Schwabhäuser's definition, the domain to be used instead of `RR` would be `NN0`, to match with the domain of words (see [df-word](https://us.metamath.org/mpeuni/df-word.html)). It would be much more convenient that `_om`.

Reply:
@tirix has responded why, and I think it's sensible.

@benjub : You said, "I think the decision is in @tirix's hands and then he/we'll adapt the text to the definition he chooses." Since this is the definition he chose, and has reaffirmed it here, I *think* this comment is resolved, so I'm resolving it. If I'm wrong please let us know :-).

Reply:
Yes, I meant NN0 sorry. So the definition should be
```
... { <. a , b >. |
E. n e. NN0 ( ( a : ( 0 ..^ n ) --> ( Base ` g ) /\ b : ( 0 ..^ n ) ->  ( Base ` g ) ) /\ ...
```

Reply:
> Let me paste Schwabäuser's definition below: <img alt="image" width="533" src="https://user-images.githubusercontent.com/5831830/216841767-bf6f2666-bfb4-4a1e-bb22-958ed162c308.png"> Which I would translate to:
> 
> > To describe more conveniently several congruences, we introduce the congruence of "geometrical figures" (made of finitely many points)
> > Definition: The n-tuples of points <a1,...,an> and <b1,...,bn> are said to be congruent when: ...
> 

The German "geometrische Figuren" could also be translated to "geometric shapes", which fits to our chosen terminology.
Furthermore, "Streckenkongruenz" does not only mean "congruence ", but "congruence of distances" - also perfectly adopted in our concepts/comments. So everything is fine now!